src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -184,7 +184,10 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
             // order for local RedirectUri
             // 1. challenge.Properties.RedirectUri
             // 2. CurrentUri if RedirectUri is not set)
-            var properties = new AuthenticationProperties(context.Properties);
+            var properties = new AuthenticationProperties(context.Properties)
+            {
+                ExpiresUtc = Options.SystemClock.UtcNow.Add(Options.CorrelationLifetime)
```

Comment:
Yep.

In practice, I can't think of a scenario where you'd like to have different lifetimes for the nonce cookie and the correlation cookie (aka `state`).

That said, `state` and `nonce` are really two different things and `nonce` is an OIDC-specific thing, while the antiforgery/correlation cookie is more general and can be applied to any protocol (even to OpenID2, that doesn't support the `state` parameter). In OIDC, using a `nonce` is not mandatory with the code flow and - unlike the correlation check - can be disabled.

So maybe it's a good thing to have a different option for the antiforgery cookie.